tests/common/test_escaping.py

```diff
@@ -24,3 +28,23 @@ def test_escape_for_bash():
 
     for string_input, string_escaped in tests:
         assert escape_for_bash(string_input) == string_escaped
+
+
+def test_escape_for_bash_double_quotes():
+    """
+    Tests various inputs for `aiida.common.escaping.escape_for_bash`.
+    double quotes tests especially useful for ENV variables with $ symbol
+    """
+
+    tests = (
+        [None, ''],
+        ['string', '"string"'],
+        ['string with space', '"string with space"'],
+        ['string with a " double quote', '''"string with a "'"'" double quote"'''],
+        [1, '"1"'],
+        [2.0, '"2.0"'],
+        ['$PWD', '"$PWD"'],
+    )
+
+    for string_input, string_escaped in tests:
+        assert escape_for_bash(string_input, use_double_quotes=True) == string_escaped
```

Comment:
This is the perfect candidate to use `pytest.mark.parametrize. I would suggest replacing the original test and the one you added with the following single test:
```suggestion
import pytest

from aiida.common.escaping import escape_for_bash


@pytest.mark.parametrize(('to_escape, expected_single_quotes, expected_double_quotes'), (
    (None, ''),
    ('string', "'string'", '"string"'),
    ('string with space', "'string with space'", '"string with space"'),
    ("string with ' single quote", """'string with '"'"' single quote'""", '''"string with "'"'" double quote"'''),
    (1, "'1'", '"1"'),
    (2.0, "'2.0'", '"2.0"'),
    ('$PWD', "'$PWD'", '"$PWD"'),
))
def test_escape_for_bash(to_escape, expected_single_quotes, expected_double_quotes):
    """Tests various inputs for `aiida.common.escaping.escape_for_bash`."""
    assert escape_for_bash(to_escape, use_double_quotes=False) == expected_single_quotes
    assert escape_for_bash(to_escape, use_double_quotes=True) == expected_double_quotes
```

Reply:
Thanks @sphuber! Absolutely it is beautiful. One line to rule all test cases 😄 

I also change the name to `environment_variables_double_quotes` as you suggest.